src/main/scala/shine/DPIA/Types/package.scala

```diff
@@ -19,13 +19,13 @@ package object Types {
   }
 
   implicit final class ArrowType[T <: PhraseType](t: T) {
-    @inline def ->: (dt: DataType): T = t
-    @inline def ->: (n: Nat): T = t
-    @inline def ->: (addressSpace: AddressSpace): T = t
-    @inline def ->: (accessType: AccessType): T = t
-    @inline def ->: (ntn: NatToNat): T = t
-    @inline def ->: (ntd: NatToData): T = t
-    @inline def ->: (unit: Unit): T = t
+    @inline def ~>: (dt: DataType): T = t
```

Comment:
Could you please explain why you changed the arrow syntax.

Reply:
Otherwise there can be ambiguities between `->:` to build a dependent function type, and `~>:` which is only used to specify dependent arguments hidden in fully applied primitives.

Reply:
I also think that this helps avoiding confusion by expressing more intent while reading the code. Are we trying to build a type, or are we just prepending constraint checks?

Reply:
Additionally, I believe that the refactoring "mistakes" that were introduced such as above in xmlPrinter come from this ambiguity. scala was probably using the wrong arrow before the changes already.

Reply:
I don't think that it's a good idea to introduce new arrow-like syntax - I don't think it is at all clear what this arrow means.

I see that you have removed the `()->` syntax and unified it with the `->:` syntax - which I guess is part of what has made this ambiguous.

I would suggest that we maybe should think about separating the checking of the types of a phrase from the type the phrase has and `t` should return.

So for something like `drop` we would change
```scala
final case class Drop(n: Nat,
                      m: Nat,
                      w: AccessType,
                      dt: DataType,
                      array: Phrase[ExpType]) extends ExpPrimitive {
  override val t: ExpType =
    (n: Nat) ~>: (m: Nat) ~>: (w: AccessType) ~>: (dt: DataType) ~>:
       (array :: expT({n + m}`.`dt, w)) ~>: expT(m`.`dt, w)
// ...
}
```
into something like
```scala
final case class Drop(n: Nat,
                      m: Nat,
                      w: AccessType,
                      dt: DataType,
                      array: Phrase[ExpType]) extends ExpPrimitive {
  array :: expT({n + m}`.`dt, w)

  override val t: ExpType = expT(m`.`dt, w)
// ...
}
```

So essentially every `:` in the current `Phrase.t` values would go and every `::` would become a statement in the class.

Opinions, @bastian-koepcke and @Bastacyclop?

Reply:
This would also emphasize that the primitives in DPIA are not functions, but represent their fully applied counterparts.

Reply:
This looks good to me :+1: I can make the changes if you want.
The only downside is not seeing the full type that we would write on paper (for the unapplied primitive function) anywhere in our code.

Reply:
I think that is fine (and could be changed in the future with the help of macros). The primitives are not functions anyway, so it's a bit confusing to have a type that suggests otherwise anyway.